### PR TITLE
doc: document uv_fs_realpath caveats

### PR DESCRIPTION
**Pull request by saghul**
Fixes: https://github.com/libuv/libuv/issues/1017

/cc @libuv/collaborators

### CODE REVIEW

File: docs/src/fs.rst [hunk 1]
```diff
@@ -288,7 +288,26 @@ API
 
 .. c:function:: int uv_fs_realpath(uv_loop_t* loop, uv_fs_t* req, const char* path, uv_fs_cb cb)
 
-    Equivalent to :man:`realpath(3)` on Unix. Windows uses ``GetFinalPathNameByHandle()``.
+    Equivalent to :man:`realpath(3)` on Unix. Windows uses `GetFinalPathNameByHandle <https://msdn.microsoft.com/en-us/library/windows/desktop/aa364962(v=vs.85).aspx>`_.
+
+    .. warning::
+        This function has certain platform specific caveats that were discovered when used in Node.
```
Comment (cjihrig) [inline docs/src/fs.rst]: The part about Node could be left out.

File: docs/src/fs.rst [hunk 2]
```diff
@@ -288,7 +288,26 @@ API
...
+          - Paths in ramdisk volumes created by tools which sidestep the Volume Manager (such as ImDisk)
+            cannot be resolved.
+          - Inconsistent casing when using drive letters.
+          - Resolved path bypasses substed drives.
+
+        While this function can still be used, it's not recommended if scenarios such as the
+        above need to be supported.
+
+        The background story and some more details on these issues can be checked
+        `here <https://github.com/nodejs/node/issues/7726>`_.
```
Comment (cjihrig) [inline docs/src/fs.rst]: Ah, I see that the part about Node above ties in with this.

**Comment by cjihrig** [comment]
LGTM

**Comment by saghul** [comment]
FTR: the reason to add the reference to Node and the meta issue is to make a stronger argument that these are not theoretical caveats, we ran into them in The Real World (TM).

File: docs/src/fs.rst [hunk 3]
```diff
@@ -288,7 +288,26 @@ API
...
+
+        * macOS and other BSDs: this function will fail with UV_ELOOP if more than 32 symlinks are
+          found while resolving the given path.  This limit is hardcoded and cannot be sidestepped.
+        * Windows: while this function works in the common case, there are a number of corner cases
+          where it doesn't:
+
+          - Paths in ramdisk volumes created by tools which sidestep the Volume Manager (such as ImDisk)
+            cannot be resolved.
+          - Inconsistent casing when using drive letters.
+          - Resolved path bypasses substed drives.
```
Comment (iWuzHere) [inline docs/src/fs.rst]: Should this be "subst'd"?

**Comment by iWuzHere** [comment]
LGTM with a question.

File: docs/src/fs.rst (hunk 3 above)
Comment (saghul) [inline docs/src/fs.rst]: right you are, sir!

**Pull request merged**